oxide-auth-rocket/src/lib.rs

```diff
@@ -169,38 +234,40 @@ impl<'r> WebResponse for OAuthResponse<'r> {
     }
 
     fn body_text(&mut self, text: &str) -> Result<(), Self::Error> {
-        self.0.set_sized_body(Cursor::new(text.to_owned()));
+        self.0.set_sized_body(text.len(),Cursor::new(text.to_owned()));
         self.0.set_header(ContentType::Plain);
         Ok(())
     }
 
     fn body_json(&mut self, data: &str) -> Result<(), Self::Error> {
-        self.0.set_sized_body(Cursor::new(data.to_owned()));
+        self.0.set_sized_body(data.len(),Cursor::new(data.to_owned()));
         self.0.set_header(ContentType::JSON);
         Ok(())
     }
 }
-
-impl<'a, 'r> FromRequest<'a, 'r> for OAuthRequest<'r> {
+#[rocket::async_trait]
+impl<'r:'static> FromRequest<'r> for OAuthRequest<'r> {
```

Comment:
This bound is confusing to me, it seems equivalent to `'r = 'static`. Wouldn't this then be only fulfilled at the substitution:

```suggestion
impl FromRequest<'static> for OAuthRequest<'static> {
```

I don't think it's very likely that this restriction is intended. In any case it could be written explicitly. Am I missing anything here?